kie-server-parent/kie-server-router/kie-server-router-proxy/src/main/java/org/kie/server/router/repository/ConfigFileWatcher.java

```diff
@@ -55,10 +58,23 @@ public ConfigFileWatcher(String configFilePath, ConfigurationMarshaller marshall
             if(toWatch.toFile().exists()) {
                 lastUpdate = Files.getLastModifiedTime(toWatch).toMillis();
             } else {
+                File file = new File(this.toWatch.toString() + "kie-server-router.json");
+                if(file.createNewFile());
+                FileOutputStream fos = new FileOutputStream(file);
+                String cfg = marshaller.marshall(configuration);
+                try (PrintWriter writer = new PrintWriter(fos)) {
```

Comment:
try (FileOutputStream fos = new FileOutputStream(file); PrintWriter writer = new PrintWriter(fos)) 

Then you can remove the close in the finally